gemini/rag-engine/rag_engine_evaluation.ipynb

```diff
@@ -0,0 +1,983 @@
+        "  - Measures the proportion of retrieved documents that are actually relevant within the top k results\n",
+        "  - Helps evaluate how \"focused\" your retrieval is\n",
+        "- **nDCG@K:**\n",
+        "  - Measures both relevance AND ranking quality\n",
+        "  - Takes into account the position of relevant documents\n",
+        "\n",
+        "Follow the Notebook to get these metrics numbers for your configurations, and to optimize your settings."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "m5sz02OdOPYk"
+      },
+      "source": [
+        "### Define evaluation helper function."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 38,
+      "metadata": {
+        "id": "VUHxSZ7hwZ6k"
+      },
+      "outputs": [],
+      "source": [
+        "def extract_rag_id(rag_resource_name):\n",
+        "    \"\"\"Extracts the RAG ID from a RAG resource name.\n",
+        "\n",
+        "    Args:\n",
+        "      rag_resource_name: The full name of the RAG resource.\n",
+        "\n",
+        "    Returns:\n",
+        "      The RAG ID extracted from the resource name.\n",
+        "    \"\"\"\n",
+        "    return rag_resource_name.split(\"/\")[-1]\n",
+        "\n",
+        "\n",
+        "def extract_doc_id(file_path):\n",
+        "    \"\"\"Extracts the document ID from a file path.\n",
+        "\n",
+        "    The function assumes the document ID is the number before the '.txt' extension in the file path.\n",
+        "\n",
+        "    Args:\n",
+        "      file_path: The path to the file.\n",
+        "\n",
+        "    Returns:\n",
+        "      The document ID extracted from the file path, or None if no ID is found.\n",
+        "    \"\"\"\n",
+        "    # Use regular expression to find the number before .txt, with optional /\n",
+        "    match = re.search(r\"/?(\\d+)\\.txt$\", file_path)\n",
+        "\n",
+        "    if match:\n",
+        "        return match.group(1)\n",
+        "    else:\n",
+        "        return None\n",
+        "\n",
+        "\n",
+        "# RAG Engine helper function to extract doc_id, snippet, and score.\n",
+        "\n",
+        "\n",
+        "def extract_retrieval_details(response):\n",
+        "    doc_id = extract_doc_id(response.source_uri)\n",
+        "    retrieved_snippet = response.text\n",
+        "    distance = response.distance\n",
+        "    return (doc_id, retrieved_snippet, distance)\n",
+        "\n",
+        "\n",
+        "# RAG Engine helper function for retrieval.\n",
+        "\n",
+        "\n",
+        "def rag_api_retrieve(query, corpus_id, top_k):\n",
+        "    return rag.retrieval_query(\n",
+        "        rag_resources=[\n",
+        "            rag.RagResource(\n",
+        "                rag_corpus=f\"projects/{PROJECT_ID}/locations/{LOCATION}/ragCorpora/{corpus_id}\"\n",
+        "            )\n",
+        "        ],\n",
```

Comment:
You should be able to just pass in the corpus id if the Vertex AI SDK was initialized with project/location

Reply:
Updated to use `rag_corpus.name` directly, also removed `extract_rag_id` function from higher up since we no longer need this.